### PR TITLE
Add option to prefer 64bit `msbuild`

### PR DESCRIPTION
**Pull request by dougbu**
### To double check:

* [x] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/core-eng/tree/master/Documentation/Validation

I'm trying this out in dotnet/aspnetcore#33497. Starting this in parallel so I don't forget between now and dotnet/aspnetcore's next Arcade update for 3.1.x.

See also conversation in that PR about _why_ this is worth having as an option.

### CODE REVIEW

**Comment by dougbu** [comment]
This fix, included in dotnet/aspnetcore#33497, built successfully in https://dev.azure.com/dnceng/public/_build/results?buildId=1189546&view=results. We seem to need this fix to build release/3.1 -- at least until we reach VS v16.9.7 or v16.10.0 (better if v16.10.2) in the regular VS2019 queues.

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -323,7 +323,16 @@ function InitializeVisualStudioMSBuild([bool]$install, [object]$vsRequirements =
   }
 
   $msbuildVersionDir = if ([int]$vsMajorVersion -lt 16) { "$vsMajorVersion.0" } else { "Current" }
-  return $global:_MSBuildExe = Join-Path $vsInstallDir "MSBuild\$msbuildVersionDir\Bin\msbuild.exe"
+
+  $local:BinFolder = Join-Path $vsInstallDir "MSBuild\$msbuildVersionDir\Bin"
+  $local:Prefer64bit = if ($vsRequirements.Prefer64bit) { $vsRequirements.Prefer64bit } else { $false }
```
Comment (MattGal) [inline eng/common/tools.ps1:328]: does this line work when vsRequirements is unset in global.json?

Comment (MattGal) [reply eng/common/tools.ps1:328]: nm  I experimented locally.

**Review by MattGal — APPROVED** [review]
LGTM

**Comment by dougbu** [comment]
@MattGal do I need to do anything special because this is targeting release/3.x❔ The change can't break anything but I'm unsure about servicing rules in this repo.

**Review by rainersigwald — APPROVED** [review]
LGTM and I would think about going so far as to flip the default in Arcade 6; VS2022 will generally use 64-bit MSBuild so it'd be nice if Arcade was consistent. We'd still need an opt-out though for tasks that are 32-bit only but don't say so correctly.

**Comment by dougbu** [comment]
Oops this is a protected branch and I can only push to 'main'. @mmitche or @MattGal please squish 'n merge

Once that's done, I don't see a backport action in this repo yet. But
https://github.com/dotnet/versions/blob/d36bff5dea12f26cb57978f5db76a0adb37d3a95/Maestro/subscriptions.json#L1157-L1166
means a subsequent merge into 'main' will contain this fix. Will just need to manually `cherry-pick` this change or the final 'main' one into 'release/5.0'.

**Comment by mmitche** [comment]
> Once that's done, I don't see a backport action in this repo yet. But
https://github.com/dotnet/versions/blob/d36bff5dea12f26cb57978f5db76a0adb37d3a95/Maestro/subscriptions.json#L1157-L1166
means a subsequent merge into 'main' will contain this fix. Will just need to manually cherry-pick this change or the final 'main' one into 'release/5.0'.

We just removed that (the common scripts don't merge cleanly at all), so it will need to be backported to main.

**Comment by dougbu** [comment]
> it will need to be backported to main

I'll get right on it.

**Pull request merged**